### ISSUE TITLE
CanvasBoost is non-functioning

### ISSUE THREAD

**Issue by javolio** [highcharts/highcharts#16727]
#### Expected behaviour
The CanvasBoost module is used when WebGL is not supported. This module should work.

#### Actual behaviour
The module generates an "Uncaught TypeError: H.eachAsync is not a function".

#### Live demo with steps to reproduce
http://jsfiddle.net/fnvdcwzo/1/

Run and observe the error message in the console.

Note that the second included script is an exact copy of the 9.3.2 Boost module, with the exception that the BoostUtils module's `hasWebGLSupport` function has been modified to always return false. This change is necessary to test the issue in browsers where that function would ordinarily return true. However, the issue happens with the unmodified Boost module in any environment where that check would return false.

#### Product version
9.3.2 (and earlier)

#### Affected browser(s)
Any without WebGL support.

#### Resolution

The resolution seems to be straightforward: eachAsync should be coming from BoostUtils, so change the erroring line from `H.eachAsync` to `butil.eachAsync`, and add BoostUtils as a dependency for the CanvasBoost module. I have tested that modification and it seems to work, but I don't know enough about this ecosystem to confidently make a PR for that change myself. Alternatively, eachAsync could be added to the Core/Globals module.

**Comment by javolio** [comment]
There's another issue that is revealed after this one has been resolved. When using the CanvasBoost module (ie running without WebGL support), if the `tooltip.shared` option is `true`, hovering over the chart errors and produces no tooltip.

Demo: http://jsfiddle.net/6tnq7oyx/

Hovering over the chart throws errors, and no tooltips are visible. If the `shared` option is set to false, things work as expected.

The second included script there is the 9.3.2 Boost module with the same modification to `hasWebGLSupport` and also the proposed solution from above.

I'm not sure how to pose that as a separate issue, since technically it's not an issue in 9.3.2 but rather in version that does not yet exist. That is, this isn't an issue until the original issue in the OP is fixed.

**Comment by raf18seb** [comment]
Hi @javolio, thank you for reporting the issues.

The first one can be easily fixed using your proposed solution.
[This line](https://github.com/highcharts/highcharts/blob/7fe23919e50a70b74def3e97ece8ebf83b34d0c0/ts/Extensions/BoostCanvas.ts#L588) should be changed from
```ts
(H as any).eachAsync(sdata, function (d: any, i: number): boolean {
...
```

to

```ts
import butils from './Boost/BoostUtils.js';
let eachAsync = butils.eachAsync;
...

eachAsync(sdata, function (d: any, i: number): boolean {
...
```

The second issue is a bit more complicated bug.

`Internal note:` seems like `chart.hoverPoint` is a full point with all its properties, but `hoverPoints` array contains boost points that lack e.g. `point.setState()` method.

**Comment by wjaspers** [comment]
Also experiencing this issue.
Of note, I noticed the `chrome://gpu` page is reporting software-only rendering for Canvas.

**Failing**
```
Canvas: Software only. Hardware acceleration disabled
Canvas out-of-process rasterization: Disabled
Compositing: Software only. Hardware acceleration disabled
Multiple Raster Threads: Disabled
Out-of-process Rasterization: Disabled
OpenGL: Disabled
Rasterization: Software only. Hardware acceleration disabled
Raw Draw: Disabled
Skia Renderer: Enabled
Video Decode: Software only. Hardware acceleration disabled
Vulkan: Disabled
WebGL: Disabled
WebGL2: Disabled
```

**Passing**
```
Canvas: Hardware accelerated
Canvas out-of-process rasterization: Disabled
Compositing: Hardware accelerated
Multiple Raster Threads: Enabled
Out-of-process Rasterization: Hardware accelerated
OpenGL: Enabled
Rasterization: Hardware accelerated
Raw Draw: Disabled
Skia Renderer: Enabled
Video Decode: Hardware accelerated
Vulkan: Disabled
WebGL: Hardware accelerated
WebGL2: Hardware accelerated
```

**Comment by vitaliy-kretsul** [comment]
Hi there guys!
Are there any plans to fix this issue? I mean the one described in start post (Uncaught TypeError: H.eachAsync is not a function)?
Thank you!

**Comment by pawellysy** [comment]
Hi @vitaliy-kretsul, sorry to inform you, but there is no ETA for this ticket yet. We will take a closer look on this bug soon.

**Comment by pawellysy** [comment]
@javolio, yes, so it seems. Thanks for the update! We will attempt fixing the other bug as well!
